core/src/main/java/org/elasticsearch/script/SimilarityScript.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import org.elasticsearch.index.similarity.ScriptedSimilarity;
+
+import java.io.IOException;
+
+/** A script that is used to build {@link ScriptedSimilarity} instances. */
+public abstract class SimilarityScript  {
+
+    /** Compute the score. */
+    public abstract double execute(double weight, ScriptedSimilarity.Query query,
```

Comment:
I think scripts would use doubles as an intermediate representation almost all the time anyway given that functions like log or sqrt produce doubles, so I felt like returning a double and casting on the elasticsearch side would save one cast from every similarity script.

Also we might switch to doubles for scores in the future (https://issues.apache.org/jira/browse/LUCENE-7517).